lib/rspec/rails/matchers/have_enqueued_mail.rb

```diff
@@ -63,20 +68,18 @@ def expected_count_message
           "#{message_expectation_modifier} #{@expected_number} #{@expected_number == 1 ? 'time' : 'times'}"
         end
 
-        def mailer_args
+        def job_match?(job)
+          MAILER_JOBS.any? { |mailer_job| send(mailer_job, job) }
```

Comment:
I think I'd rather we just called `legacy_mail?(mailer_job, job) || parameterized_mail?(mailer_job, job) || unified_mail?(mailer_job, job)` rather than constantising it and iterating.

Reply:
sure, updated.. I used constant because I thought it would be easier later IF we somehow need to add check for other job..